core/src/utilities/helpers/storage-helper.js

```diff
@@ -0,0 +1,154 @@
+import { IframeHelpers } from './iframe-helpers';
+
+class StorageHelperClass {
+  constructor() {
+    this.init=false;
+    this.storage=undefined;
+    this.browseSupported=undefined;
+  }
+
+  checkInit(){
+    if (this.init){
+      return;
+    }
+    this.storage = window.localStorage
+    this.browseSupported = this.supportLocalStorage();
+    this.init=true;
+  }
+
+  supportLocalStorage(){
+    try {
+      return 'localStorage' in window && window['localStorage'] !== null;
+    } catch(e) {
+      return false;
+    }
+  }
+
+  checkStorageBrowserSupport(){
+    if (!this.browseSupported){
+      throw "Browser does not support local storage"
+    }
+  }
+
+  process(microfrontendId, hostname, id, operation, params){
+    try{
+        this.checkInit();
+        this.checkStorageBrowserSupport();
+        const operationFunction = this[operation];
+        if (typeof operationFunction !== 'function'){
+          throw operation + " is not a supported operation for the storage";
+        }
+        const result = operationFunction.bind(this, hostname, params)();
+        //const result = operationFunction(microfrontendId, params);
+        this.sendBackOperation(microfrontendId, id, 'OK', result);
+      }catch(error){
+        console.log(error);
+        this.sendBackOperation(microfrontendId, id, 'ERROR', error);
+      }
+  }
+
+  setItem(hostname, params) {
+    this.checkKey(params);
+    const value = this.stringifyValue(params.value);
+    const key =  "Luigi#"+hostname + "#"+ params.key.trim();
```

Comment:
maybe key generation can be extracted to a function ("Luigi#"+hostname + "#") since it is used several times

Reply:
Done